mPower2/MotorControl/StepObjects/MCTHandSelectionStepObject.swift

```diff
@@ -0,0 +1,116 @@
+//  MotorControl
+//
+//  Copyright © 2018 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/// A Subclass of RSDFormUIStepObject which uses MCTHandSelectionDataSource.
+public class MCTHandSelectionStepObject : RSDFormUIStepObject {
+    override public func instantiateDataSource(with taskPath: RSDTaskPath, for supportedHints: Set<RSDFormUIHint>) -> RSDTableDataSource? {
+        return MCTHandSelectionDataSource(step: self, taskPath: taskPath, supportedHints: supportedHints)
+    }
+}
+
+/// An enum for representing the different possible usable hands a user may have.
+public enum MCTHandSelection : String, Codable {
+    case left, right, both
+}
+
+/// The object that serves as the data soruce for an MCTHandSelectionStep
+public class MCTHandSelectionDataSource : RSDFormStepDataSourceObject {
+    
+    /// Key for the randomized hand order in the task result.
+    public static let handOrderKey = "handOrder"
+    
+    /// Key for which hands the user said they could use in the task result.
+    public static let selectionKey = "handSelection"
+    
+    /// Override the initial result to look for the user's previous answer to this quesiton in
+    /// UserDefaults.
+    override open var initialResult : RSDCollectionResult? {
+        /// TODO rkolmos 04/10/2018 fix this to not use string builders
```

Comment:
Using the string builder is fine so long as it is **not** an enum that uses the default `description` property on the object to set the string value. Here you are okay b/c the identifier is a `String` and not a `RawRepresentable`.